### PR TITLE
initialise work-queues: a re-submission of the same patches, sent to alsa-devel / sof

### PR DESCRIPTION
**Pull request by lyakh**
as suggested by Mark and requested by @ranj063 these are now two separate patches.

### CODE REVIEW

File: sound/soc/sof/topology.c [hunk 1]
```diff
@@ -2541,7 +2544,7 @@ static int sof_dai_load(struct snd_soc_component *scomp, int index,
 		return ret;
 
 	dev_vdbg(scomp->dev, "tplg: pcm %s stream tokens: capture d0i3:%d\n",
-		 spcm->pcm.pcm_name, spcm->stream[1].d0i3_compatible);
+		 spcm->pcm.pcm_name, spcm->stream[stream].d0i3_compatible);
```
Comment (kv2019i) [inline sound/soc/sof/topology.c:2552]: @lyakh stream = SNDRV_PCM_STREAM_CAPTURE ?

Comment (ranj063) [reply sound/soc/sof/topology.c:2552]: @kv2019i this is done on line 2540

Comment (kv2019i) [reply sound/soc/sof/topology.c:2552]: Aa, my bad, then it's ok.

**Comment by kv2019i** [comment]
@lyakh So do we put this merged via here or is the upstream submission still alive?

**Comment by lyakh** [comment]
> @lyakh So do we put this merged via here or is the upstream submission still alive?

@kv2019i IIUC merging locally would now be the preferred way. If you do so, I'll email alsa-devel / Mark to not merge it via his tree.

**Comment by kv2019i** [comment]
@lyakh wrote:
> @kv2019i IIUC merging locally would now be the preferred way. If you do so, I'll email alsa-devel / Mark to not merge it via his tree.

Ack, done.

**Pull request merged**